src/services/github.ts

```diff
@@ -95,79 +102,74 @@ const fetchBaseTreeSha = async commitSha => {
   }
 }
 
-export const createTree = async (baseTreSha, fileContent, chainId) => {
-  const endpoint = `${ENDPOINT_BASE}/git/trees`
-  const network = getNetworkName(chainId).toLowerCase()
-
-  const bodyData = {
-    base_tree: baseTreSha,
-    tree: [
-      {
-        path: `${network}.json`,
-        mode: '100644',
-        type: 'blob',
-        content: JSON.stringify(fileContent, null, 4),
-      },
-    ],
-  }
-  try {
-    const result = await fetch(endpoint, {
-      method: 'POST',
-      headers: {
-        Authorization: `token ${GITHUB_API_TOKEN}`,
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify(bodyData),
-    })
-    const data = await result.json()
-
-    return { data: data.sha, error: !result.ok }
-  } catch (err) {
-    console.error(`Error requesting tree sha`, err)
-    return { error: true }
-  }
+type TreeCommitParams = {
+  baseTreSha: string
+  latestCommitSha: string
+  daoMetadataName: string
+  newContent: any
+  networkName: string
+  ownerRepo: string
+  repo: string
 }
 
-const createCommit = async (latestCommitSha, newTreeSha, daoName) => {
-  const endpoint = `${ENDPOINT_BASE}/git/commits`
+export const createTreeCommitAndChangeHeads = async ({
+  baseTreSha,
+  latestCommitSha,
+  daoMetadataName,
+  newContent,
+  networkName,
+  ownerRepo,
+  repo,
+}: TreeCommitParams) => {
+  const endpoint = `${MIDDLEWARE_ENDPOINT}/v1/dao`
+
   const bodyData = {
-    parents: [latestCommitSha],
-    tree: newTreeSha,
-    message: ` ${daoName} added`,
```

Comment:
Don't you forget this variable? The `newTreeSha` is not being used after.

Reply:
Idk if i understand your question but that functions was migrated to backend

https://github.com/kamikazebr/dao-list-middleware/blob/master/src/routes/daolist.js#L154

Check that here.

Reply:
Yep, my mistake. It's being used yes.